protocol/messenger_communities.go

```diff
@@ -635,6 +658,74 @@ func (m *Messenger) handleCommunityEncryptionKeysRequest(community *communities.
 	return nil
 }
 
+func (m *Messenger) handleCommunitySharedAddressesRequest(state *ReceivedMessageState, community *communities.Community, signer *ecdsa.PublicKey) error {
+	if !community.HasMember(signer) {
+		return communities.ErrMemberNotFound
+	}
+
+	pkStr := common.PubkeyToHex(signer)
+
+	revealedAccounts, err := m.communitiesManager.GetRevealedAddresses(community.ID(), pkStr)
+	if err != nil {
+		return err
+	}
+
+	usersSharedAddressesProto := &protobuf.CommunitySharedAddressesResponse{
+		CommunityId:      community.ID(),
+		RevealedAccounts: revealedAccounts,
+	}
+
+	payload, err := proto.Marshal(usersSharedAddressesProto)
+	if err != nil {
+		return err
+	}
+
+	rawMessage := common.RawMessage{
+		Payload:             payload,
+		Sender:              community.PrivateKey(),
+		CommunityID:         community.ID(),
+		SkipEncryptionLayer: true,
+		MessageType:         protobuf.ApplicationMetadataMessage_COMMUNITY_SHARED_ADDRESSES_RESPONSE,
+		PubsubTopic:         shard.DefaultNonProtectedPubsubTopic(),
+		ResendType:          common.ResendTypeRawMessage,
+		ResendMethod:        common.ResendMethodSendPrivate,
+		Recipients:          []*ecdsa.PublicKey{signer},
+	}
+
+	_, err = m.sender.SendPrivate(context.Background(), signer, &rawMessage)
+	if err != nil {
+		return err
+	}
+
+	// Mark Backup as handled to tell the test that we handled the request
+	// This is a bit of a hack since no backup was performed, but it does come from someone's backup
+	state.Response.BackupHandled = true
```

Comment:
I am opting for not commiting/merging this hack :police_officer:

Reply:
~~ACAB~~ ok officer :upside_down_face: